spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -237,6 +195,25 @@ public void stopJobExecution(long jobExecutionId)
 	}
 
 	private TaskJobExecution getTaskJobExecution(JobExecution jobExecution) {
+		return new TaskJobExecution(getTaskExecutionId(jobExecution), jobExecution,
+				isTaskDefined(jobExecution), jobExecution.getStepExecutions().size());
+	}
+
+	private List<TaskJobExecution> getTaskJobExecutionsWithStepCountForList(Collection<JobExecution> jobExecutions) {
+		Assert.notNull(jobExecutions, "jobExecutions must not be null");
+		List<TaskJobExecution> taskJobExecutions = new ArrayList<>();
+		for (JobExecution jobExecution : jobExecutions) {
+			taskJobExecutions.add(getTaskJobExecutionWithStepCount(jobExecution));
```

Comment:
This could potentially become performance bottleneck as we issue a JDBC query for **each** job execution to get the step execution count. Maybe we could do some joins at the `list` query for job executions to get the step execution count?

Reply:
Updated.   Retrieves count directly from the JobExecutionStepCount instance